android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java

```diff
@@ -272,6 +273,7 @@ public void setupDialog(@NonNull Dialog dialog, int style) {
                                         selectedNetwork, accounts, filterByTokens, false,
                                         (assetPrices, fullTokenList, nativeAssetsBalances,
                                                 blockchainTokensBalances) -> {
+                                            if (!canUpdateUi()) return;
```

Comment:
isn't it sufficient only one check? as that one could only occur only when first is passed.

Reply:
Both calls need to be safeguarded because the dialog can be dismissed anytime, either after the first or after the second.